### DISCUSSION TITLE
Issue with Enhanced Table Plugin Not Showing in Kibana 8.4 Visualizations Menu

### DISCUSSION THREAD

**Discussion by HesusG** [fbaligand/kibana-enhanced-table#339] [Q&A]
I'm experiencing an issue with the Enhanced Table plugin (version 1.13.3) on Kibana 8.4. I followed the installation process as documented, and the plugin seems to install correctly. I can confirm that it's listed when I use the plugin list command in my shell. However, I'm facing a problem where the Enhanced Table does not appear in the Visualizations menu in Kibana.

There are no error messages when installing the plugin. 
![image](https://github.com/fbaligand/kibana-enhanced-table/assets/69993487/0edb58b0-63a1-4899-a011-2f66eaa377fd)

**Comment by fbaligand** [comment]
Well, since few Kibana versions, you have to do these step to see "Enhanced Table" and "Document Table" visualizations:
- first, go to "Visualize Library" app
- then click on "Create visualization" button
- and here, click on "Aggregation based" category
- and only there, you should see "Enhanced Table" and "Document Table" visualizations
- if you don't see it there, so do "Shift+F5" in your browser to force browser cache refresh

**Reply by OskRgh** [reply]
hi Fabien,
I'm experiencing HesusG's behavior too.
I'm trying to install the plugin in version 8.14.3 (downloaded **_enhanced-table-1.14.0_8.14.3.zip_**) but no matter what I do (followed the installation and first use steps) I still cannot see "Enhanced Table" and/or "Document Table" in the list of "Aggregation based" visualizations.
Any idea what the problem might be or any suggestion to troubleshoot the problem?
Thanks in advance for your help!

**Reply by fbaligand** [reply]
First, some questions:
- did you install this package on Kibana 8.14.3 precisely?
- did you restart Kibana after install?
- Did you try Shift+F5 in your browser to force page and cache reload?

**Reply by OskRgh** [reply]
hi Fabien,

The answer is Yes to the 3 questions.

This is the output from a terminal window:

[root@xxx bin]# cd /usr/share/kibana/bin
[root@xxx bin]# ./kibana --version | grep serviceVersion
{"log.level":"info","@timestamp":"2024-09-18T19:17:54.530Z","log.logger":"elastic-apm-node","ecs.version":"8.10.0","agentVersion":"4.5.0","env":{"pid":1931,"proctitle":"./../node/bin/node","os":"linux 3.10.0-1160.71.1.el7.x86_64","arch":"x64","host":"MALCPUBVEST","timezone":"UTC+0200","runtime":"Node.js v20.13.1"},"config":{"active":{"source":"start","value":true},"breakdownMetrics":{"source":"start","value":false},"captureBody":{"source":"start","value":"off","commonName":"capture_body"},"captureHeaders":{"source":"start","value":false},"centralConfig":{"source":"start","value":false},"contextPropagationOnly":{"source":"start","value":true},"environment":{"source":"start","value":"production"},"globalLabels":{"source":"start","value":[["kibana_uuid","968b33b8-9fa0-431b-b71d-a588c2469b6a"],["git_rev","465f50087cd040ef03e6ccec1cb7737427a713ce"]],"sourceValue":{"kibana_uuid":"968b33b8-9fa0-431b-b71d-a588c2469b6a","git_rev":"465f50087cd040ef03e6ccec1cb7737427a713ce"}},"logLevel":{"source":"default","value":"info","commonName":"log_level"},"metricsInterval":{"source":"start","value":120,"sourceValue":"120s"},"serverUrl":{"source":"start","value":"https://kibana-cloud-apm.apm.us-east-1.aws.found.io/","commonName":"server_url"},"transactionSampleRate":{"source":"start","value":0.1,"commonName":"transaction_sample_rate"},"captureSpanStackTraces":{"source":"start","sourceValue":false},"secretToken":{"source":"start","value":"[REDACTED]","commonName":"secret_token"},"serviceName":{"source":"start","value":"kibana","commonName":"service_name"},"serviceVersion":{"source":"start",**"value":"8.14.3"**,"commonName":"service_version"}},"activationMethod":"require","message":"Elastic APM Node.js Agent v4.5.0"}
Kibana should not be run as root.  Use --allow-root to continue.
[root@xxx bin]# ./kibana-plugin list
**enhancedTable@1.14.0**
[root@xxx bin]#

When I browse to the Visualization app there's no trace of "Enhanced Table" and/or "Document Table". Please see attached file
![NewVisualizationWindow](https://github.com/user-attachments/assets/b69d1673-12e9-4e01-9be8-e0428be006de)

It might be worth mentioning that this is part of a migration effort from a 7.x.y version to 8.14.3 ... but I have detected no other problems. Dashboards and such working correctly except for those using the "Enhanced Table" plugin... those display a "Invalid visualization type "enhanced-table"" as if the plugin has not been installed.
 While migrating I got the following message:
[ERROR][plugins-service] Error: Plugin "enhancedTable" is only compatible with Kibana version "7.17.15", but used Kibana version is "8.14.3". (incompatible-version, /usr/share/kibana/plugins/enhanced-table/kibana.json)
So I upgraded the plugin removing the old one and installing the new one as described in my previous message (from file enhanced-table-1.14.0_8.14.3.zip)

I have tried all sorts of things... I just re-installed it and this is what I get (seems everything went OK):

[root@xxx bin]# ./kibana-plugin remove enhancedTable
Removing enhancedTable...
Plugin removal complete
[root@xxx bin]# ./kibana-plugin install file:///home/elasticadmin/instaladores/enhanced-table-1.14.0_8.14.3.zip
Attempting to transfer from file:///home/elasticadmin/instaladores/enhanced-table-1.14.0_8.14.3.zip
Transferring 1083040 bytes....................
Transfer complete
Retrieving metadata from plugin archive
Extracting plugin archive
Extraction complete
Plugin installation complete
[root@xxx bin]# ./kibana-plugin list
enhancedTable@1.14.0
[root@xxx bin]#

I can see a /usr/share/kibana/plugins/enhancedTable has been created

Please let me know of any information I can provide to help troubleshoot the problem and, again, thanks so much!
